### ISSUE TITLE
Neural Network Fundamentals and Backpropagation

### ISSUE THREAD

**Issue by Rishikesh75** [Rishikesh75/backend-devops-lab#17]
## Parent Epic
Part of #9 — Deep Learning Interview Preparation.

## Topics
- Neurons and layers
- Activations
- Forward propagation
- Backpropagation
- Computational graphs
- Gradient descent
- Vanishing/exploding gradients

## Interview Focus
Explain the complete forward/backward pass and why gradients are needed. Include common interview follow-ups.
